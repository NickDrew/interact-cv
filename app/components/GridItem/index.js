/**
 *
 * GridItem
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';
import is from 'styled-is';
const GridItem = styled.div`
  ${is('column')`grid-column: ${props => props.column}`}
  ${is('row')`grid-row: ${props => props.row}`}
  ${is('area')`grid-area: ${props => props.area}`}`;

GridItem.propTypes = {
  column: PropTypes.string,
  row: PropTypes.string,
  area: PropTypes.string,
};

export default GridItem;
