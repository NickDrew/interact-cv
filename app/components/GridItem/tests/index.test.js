import React from 'react';
import { render, getByTestId } from 'react-testing-library';
import 'jest-dom/extend-expect'; // add some helpful assertions

import GridItem from '../index';

describe('<GridItem />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<GridItem />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to adopt passed grid-column style', () => {
    const id = 'test';
    const col = '1/5';
    const { container } = render(<GridItem data-testid={id} column={col} />);

    expect(getByTestId(container, id)).toHaveStyle(`grid-column: ${col}`);
  });

  it('Expect to adopt passed grid-row style', () => {
    const id = 'test';
    const row = '3';
    const { container } = render(<GridItem data-testid={id} row={row} />);

    expect(getByTestId(container, id)).toHaveStyle(`grid-row: ${row}`);
  });

  it('Expect to adopt passed grid-area style', () => {
    const id = 'test';
    const area = '2 / 1 / span 2 / span 3';
    const { container } = render(<GridItem data-testid={id} area={area} />);

    expect(getByTestId(container, id)).toHaveStyle(`grid-area: ${area}`);
  });

  it('Should render and match the snapshot', () => {
    const col = '1/5';
    const row = '3';
    const {
      container: { firstChild },
    } = render(<GridItem column={col} row={row} />);
    expect(firstChild).toMatchSnapshot();
  });
});
