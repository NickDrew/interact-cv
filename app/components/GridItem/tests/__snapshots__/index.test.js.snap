// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItem /> Should render and match the snapshot 1`] = `
<div
  class="GridItem-sc-1m27xx7-0 kPRoTX"
/>
`;
